android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/DemosFragment.java

```diff
@@ -0,0 +1,110 @@
+package org.physical_web.physicalweb;
+
+import android.app.ListFragment;
+import android.content.ContentResolver;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+
+/**
+ * This class shows a list of demos.
+ */
+public class DemosFragment extends ListFragment {
+  private static final String TAG = DemosFragment.class.getSimpleName();
+  private DemosAdapter mAdapter;
+
+  @Override
+  public View onCreateView(LayoutInflater layoutInflater, ViewGroup container,
+      Bundle savedInstanceState) {
+    getActivity().getActionBar().setDisplayHomeAsUpEnabled(true);
+    View rootView = layoutInflater.inflate(R.layout.fragment_demos, container, false);
+    mAdapter = new DemosAdapter();
+    setListAdapter(mAdapter);
+    return rootView;
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    getActivity().getActionBar().setTitle(R.string.demos);
+  }
+
+  @Override
+  public void onPrepareOptionsMenu(Menu menu) {
+    super.onPrepareOptionsMenu(menu);
+    Utils.hideAllMenuItems(menu);
+  }
+
+  @Override
+  public void onListItemClick(ListView l, View v, int position, long id) {
+    mAdapter.getItem(position).startDemo();
+  }
+
+  private class DemosAdapter extends BaseAdapter {
+    Demo[] demos;
+
+    public DemosAdapter() {
+      demos = new Demo[]{
+          new Demo() {
```

Comment:
Could you go ahead and make:
* org.physical_web.physicalweb.demos.Demo
* org.physical_web.physicalweb.demos.FatBeaconHelloWorld

And then just register it in the onCreate of this fragment.